src/synthcity/plugins/core/models/RGCNConv.py

```diff
@@ -0,0 +1,297 @@
+    Edge type needs to be a one-dimensional :obj:`torch.long` tensor which
+    stores a relation identifier
+    :math:`\in \{ 0, \ldots, |\mathcal{R}| - 1\}` for each edge.
+    .. note::
+        This implementation is as memory-efficient as possible by iterating
+        over each individual relation type.
+        Therefore, it may result in low GPU utilization in case the graph has a
+        large number of relations.
+        As an alternative approach, :class:`FastRGCNConv` does not iterate over
+        each individual type, but may consume a large amount of memory to
+        compensate.
+        We advise to check out both implementations to see which one fits your
+        needs.
+    Args:
+        in_channels (int or tuple): Size of each input sample. A tuple
+            corresponds to the sizes of source and target dimensionalities.
+            In case no input features are given, this argument should
+            correspond to the number of nodes in your graph.
+        out_channels (int): Size of each output sample.
+        num_relations (int): Number of relations.
+        num_bases (int, optional): If set, this layer will use the
+            basis-decomposition regularization scheme where :obj:`num_bases`
+            denotes the number of bases to use. (default: :obj:`None`)
+        num_blocks (int, optional): If set, this layer will use the
+            block-diagonal-decomposition regularization scheme where
+            :obj:`num_blocks` denotes the number of blocks to use.
+            (default: :obj:`None`)
+        aggr (string, optional): The aggregation scheme to use
+            (:obj:`"add"`, :obj:`"mean"`, :obj:`"max"`).
+            (default: :obj:`"mean"`)
+        root_weight (bool, optional): If set to :obj:`False`, the layer will
+            not add transformed root node features to the output.
+            (default: :obj:`True`)
+        is_sorted (bool, optional): If set to :obj:`True`, assumes that
+            :obj:`edge_index` is sorted by :obj:`edge_type`. This avoids
+            internal re-sorting of the data and can improve runtime and memory
+            efficiency. (default: :obj:`False`)
+        bias (bool, optional): If set to :obj:`False`, the layer will not learn
+            an additive bias. (default: :obj:`True`)
+        **kwargs (optional): Additional arguments of
+            :class:`torch_geometric.nn.conv.MessagePassing`.
+    """
+
+    def __init__(
+        self,
+        in_channels: Union[int, Tuple[int, int]],
+        out_channels: int,
+        num_relations: int,
+        num_bases: Optional[int] = None,
+        num_blocks: Optional[int] = None,
+        aggr: str = "mean",
+        root_weight: bool = True,
+        is_sorted: bool = False,
+        bias: bool = True,
+        **kwargs: Any,
+    ) -> None:
+        kwargs.setdefault("aggr", aggr)
+        super().__init__(node_dim=0, **kwargs)
```

Comment:
Is this `node_dim` a parameter or needs to be hardcoded to 3?

Reply:
I have exposed it as a parameter now.